ciao-controller/compute.go

```diff
@@ -906,7 +911,16 @@ func createServer(w http.ResponseWriter, r *http.Request, context *controller) {
 	}
 	servers.TotalServers = len(instances)
 
-	b, err := json.Marshal(servers)
+	// set machine ID for OpenStack compatibility
+	server.Server.ID = instances[0].ID
```

Comment:
I see, then the verification should be modified  in  https://github.com/01org/ciao/blob/master/ciao-controller/compute.go#L881  instead of "!= 0" we shoud have  "> 0" and it would prevent the failure